gwcs/wcs.py

```diff
@@ -1757,21 +1780,12 @@ def _fit_2D_poly(ntransform, npoints, degree, max_error,
     return fit_poly_x, fit_poly_y, max_resid
 
 
-def _make_sampling_grid(npoints, bounding_box):
-    (xmin, xmax), (ymin, ymax) = bounding_box
-    xsize = xmax - xmin
-    ysize = ymax - ymin
-    crpix1 = int(xsize / 2)
-    crpix2 = int(ysize / 2)
-    stepsize_x = int(xsize / npoints)
-    stepsize_y = int(ysize / npoints)
-    # Ensure last row and column are part of the evaluation grid.
-    y, x = np.mgrid[: ymax + 1: stepsize_y, : xmax + 1: stepsize_x]
-    x[:, -1] = xsize - 1
-    y[-1, :] = ysize - 1
-    u = x - crpix1
-    v = y - crpix2
-    return u, v
+def _make_sampling_grid(npoints, bounding_box, crpix):
+    from .wcstools import grid_from_bounding_box
```

Comment:
Any reason this import shouldn't be at the top?

Reply:
Yes. Unfortunately `.wcstools` imports from `.wcs`. Placing this import here avoids circular imports.

Reply:
You may want to add a comment here to indicate why it is imported inside the function.

```python
# Import here to avoid circular imports
from .wcstools import grid_from_bounding_box
```

So that a future busybody (I resemble that) doesn't try to change it.

Reply:
Or a clueless reviewer like me. :-)